### PR TITLE
Rever

### PR DESCRIPTION
**Pull request by scopatz**
Hello All, 

Along with the 0.6.0 release that I just pushed out, I added [rever](https://regro.github.io/rever-docs/index.html) support to xonsh.  Rever is a new project for managing software releases.  This was inspired by xonsh's `release.xsh` script and rever is meant to replace it.  

Rever itself is, of course, written in xonsh :smile: 

Please let me know if you have any questions or concerns.  CC @CJ-Wright

### CODE REVIEW

File: rever.xsh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    ]
+$CHANGELOG_FILENAME = 'CHANGELOG.rst'
+$CHANGELOG_TEMPLATE = 'TEMPLATE.rst'
+
+#$PYTEST_COMMAND = "rm -r .cache/ __pycache__/ */__pycache__ */*/__pycache__ && pytest"
+
+$TAG_REMOTE = 'git@github.com:xonsh/xonsh.git'
+$TAG_TARGET = 'master'
+
+$GHPAGES_REPO = 'git@github.com:scopatz/xonsh-docs.git'
```
Comment (CJ-Wright) [inline rever.xsh]: Should this eventually go under xonsh the org?

Comment (scopatz) [reply rever.xsh]: Yes. It mostly is, but we haven't switched the CNAME file over yet.

Comment (CJ-Wright) [reply rever.xsh]: ah ok

**Comment by scopatz** [comment]
ping!

File: rever.xsh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+               ]
+
+$VERSION_BUMP_PATTERNS = [
+    ('.appveyor.yml', 'version:.*', 'version: $VERSION.{build}'),
+    ('xonsh/__init__.py', '__version__\s*=.*', "__version__ = '$VERSION'"),
+    ]
+$CHANGELOG_FILENAME = 'CHANGELOG.rst'
+$CHANGELOG_TEMPLATE = 'TEMPLATE.rst'
+
+#$PYTEST_COMMAND = "rm -r .cache/ __pycache__/ */__pycache__ */*/__pycache__ && pytest"
```
Comment (gforsyth) [inline rever.xsh]: Non-blocking, but should this just be removed?

Comment (scopatz) [reply rever.xsh]: Doh! Yes!

Comment (scopatz) [reply rever.xsh]: Done!

**Comment by gforsyth** [comment]
Hey @scopatz -- looks good and I really like the look of `rever`.  One issue that I think is probably specific to the xonsh docs, but the release seems to have added the `devN` suffix to the version when it published the `0.6.0` docs, (`xonsh 0.6.0.dev151 documentation` on xon.sh)

We definitely don't have 151 commits since the 0.6.0 release, but I think that _is_ the number of commits between 0.5.12 and 0.6.0.

**Comment by scopatz** [comment]
> I really like the look of rever

Thanks!  Of course you and the rest of xore are all welcome to join / participate in rever and the other regro projects. We even have a fancy front page now :) https://regro.github.io/ 

> the release seems to have added the devN suffix to the version when it published the 0.6.0 docs

Thanks for pointing that out.  I think that might be a rever bug / a bug with the way we are using rever.  IIRC that commit number comes from the diff between the current state of the repo and the last tag.  We are building the docs before we tag in this setup because I wanted to be sure we could build the docs before tagging the release.  Maybe we should rearrange this or build in a release valve to the install to avoid the devN?  Either way, I think that is fine tuning :)

**Comment by gforsyth** [comment]
Agreed. Merging

**Pull request merged**